pyuvdata/uvdata/mir_meta_data.py

```diff
@@ -0,0 +1,3087 @@
+                                break
+
+        # If you've gotten to this point and you still have unresolved overlap
+        # entries, then we have a problem -- time to raise an error.
+        if (idx_count - (len(index_dict1) + len(index_dict2))) != len(key_overlap):
+            raise MirMetaError(
+                "Cannot combine objects, as both contain overlapping index markers "
+                "with different metadata. You can bypass this error by setting "
+                "overwrite=True."
+            )
+
+        this_idx = sorted(index_dict1.values())
+        other_idx = sorted(index_dict2.values())
+
+        return this_idx, other_idx, this_mask[this_idx], other_mask[other_idx]
+
+    def fromfile(self, filepath):
+        """
+        Read in data for a MirMetaData object from disk.
+
+        Parameters
+        ----------
+        filepath : str
+            Path of the folder containing the metadata in question.
+        """
+        if self._binary_dtype is None:
+            self._data = np.fromfile(
+                os.path.join(filepath, self._filetype),
+                dtype=self.dtype,
+            )
+        else:
+            self._data = np.fromfile(
+                os.path.join(filepath, self._filetype), dtype=self._binary_dtype
+            ).astype(self.dtype)
+
+        self._mask = np.ones(len(self), dtype=bool)
+        self._set_header_key_index_dict()
+
+    def _writefile(self, filepath, append_data, datamask=...):
+        """
+        Write _data attribute to disk.
+
+        This function is part of the low-level API, which is called when calling the
+        `tofile` method. It is broken out separately here to enable subclasses to
+        differently specify how data are written out (namely binary vs text).
+
+        Parameters
+        ----------
+        filepath : str
+            Path of the folder to write the metadata into.
+        append_data : bool
+            If set to True, will append to an existing file, otherwise the method will
+            overwrite any previously written data.
+        datamask : ndarray of bool
+            Mask for selecting which data to write to disk. Default is the entire
+            array of metadata.
+        """
+        with open(filepath, "ab" if append_data else "wb+") as file:
+            if self._binary_dtype is None:
+                self._data[datamask].tofile(file)
+            else:
+                self._data[datamask].astype(self._binary_dtype).tofile(file)
+
+    def tofile(
+        self,
+        filepath,
+        overwrite=False,
+        append_data=False,
+        check_index=False,
+    ):
+        """
+        Write a metadata file to disk.
+
+        This function is part of the low-level API, which is called when calling the
+        `tofile` method. It is broken out separately here to enable subclasses to
+        differently specify how data are written out (namely binary vs text).
```

Comment:
I think this paragraph doesn't actually apply to this method.

Reply:
Removed in a2f46b4b.